pkg/kubelet/prober/prober.go

```diff
@@ -219,16 +219,25 @@ type execInContainer struct {
 	run func() ([]byte, error)
 }
 
-func (p *prober) newExecInContainer(container api.Container, containerID kubecontainer.ContainerID, cmd []string) exec.Cmd {
+func (p *prober) newExecInContainer(container api.Container, containerID kubecontainer.ContainerID, cmd []string, timeout time.Duration) exec.Cmd {
 	return execInContainer{func() ([]byte, error) {
 		var buffer bytes.Buffer
 		output := ioutils.WriteCloserWrapper(&buffer)
-		err := p.runner.ExecInContainer(containerID, cmd, nil, output, output, false)
-		if err != nil {
-			return nil, err
+		errCh := make(chan error, 1)
```

Comment:
`defer close(errCh)` will close the channel when the function returns. If the goroutine below is still running, we'll have a panic in that goroutine because we can't send on a closed channel.
https://golang.org/ref/spec#Send_statements